File: src/my_project/interviews/interview_1_msft/sort_colors.py
from typing import List, Union 
from abc import ABC, abstractmethod

class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        c0=c1=c2=0
        for num in nums: 
            if num==0:
                c0+=1
            elif num==1:
                c1+=1
            elif num==2:
                c2+=1

        nums[0:c0] = [0]*c0
        nums[c0:c0+c1] = [1]*c1 
        nums[c0+c1:c0+c1+c2] = [2]*c2 

        return nums
        

        pass 

File: my_project_typescript/medium_problems_typescript/from1to50/sortColors.ts
function sortColors(nums: number[]): void {

    let c0: number = 0
    let c1: number = 0
    let c2: number = 0
    

    for (let i = 0; i < nums.length; i++){
        if (nums[i] === 0){
            c0 += 1
        }
        else if (nums[i] === 1){
            c1 += 1
        }
        else {
            c2 += 1
        }
    }

    for (let i = 0; i < c0; i++ ){
        nums[i] = 0
    }
    for (let i = c0; i < c0+c1; i++ ){
        nums[i] = 1
    }
    for (let i = c0+c1; i < nums.length; i++ ){
        nums[i] = 2
    }




};